### PR TITLE
Some fixes and feature requests

### PR DESCRIPTION
**Pull request by sarthakpati**
Fixes issue #829, #848, #894, #892, #911

Includes #895, #893

### CODE REVIEW

**Comment by sarthakpati** [comment]
This is ready for review, @CBICA/devteam.

File: src/applications/Preprocessing/src/Preprocessing.cxx [hunk 1]
```diff
@@ -75,36 +78,121 @@ int algorithmsRunner()
...
+      auto inputImageFiles = cbica::stringSplit(inputImageFile, ",");
+      std::vector< typename TImageType::Pointer > inputImages, inputMask;
+      for (size_t i = 0; i < inputImageFiles.size(); i++)
+      {
+        auto currentImage = cbica::ReadImage< TImageType >(inputImageFiles[i]);
+        inputImages.push_back(currentImage);
+        //if (!inputMaskFile.empty())
+        //{
+        //  inputMask.push_back(cbica::ReadImage< TImageType >(inputMaskFile));
+        //}
```
Comment (ashishsingh18) [inline src/applications/Preprocessing/src/Preprocessing.cxx]: Please delete commented code.

File: src/applications/Preprocessing/src/Preprocessing.cxx [hunk 2]
```diff
@@ -75,36 +78,121 @@ int algorithmsRunner()
...
+      }
+
+      using NewImageType = itk::Image< typename TImageType::PixelType, TImageType::ImageDimension + 1 >;
+      auto combinedImage = cbica::GetJoinedImage< TImageType, NewImageType >(inputImages);
+      //auto combinedMask = combinedImage;
+      //if (!inputMask.empty())
+      //{
+      //  combinedMask = cbica::GetJoinedImage< TImageType, NewImageType >(inputMask);
+      //}
+
```
Comment (ashishsingh18) [inline src/applications/Preprocessing/src/Preprocessing.cxx]: Please delete commented code.

File: src/applications/Preprocessing/src/Preprocessing.cxx (hunk 1 above)
Comment (sarthakpati) [reply src/applications/Preprocessing/src/Preprocessing.cxx]: The is commented for potential future expansion.

Comment (sarthakpati) [reply src/applications/Preprocessing/src/Preprocessing.cxx]: The is commented for potential future expansion.

Comment (sarthakpati) [reply src/applications/Preprocessing/src/Preprocessing.cxx]: The is commented for potential future expansion.

File: src/applications/Preprocessing/src/Preprocessing.cxx (hunk 2 above)
Comment (sarthakpati) [reply src/applications/Preprocessing/src/Preprocessing.cxx]: The is commented for potential future expansion.

**Comment by sarthakpati** [comment]
DICOM I/O is handled at the ```cbica::ReadImage``` level. If that works as expected, everything else does, as well.

**Comment by sarthakpati** [comment]
DICOM I/O handling has been updated courtesy https://github.com/sarthakpati/CaPTk/commit/207bbbaedfd4bb5c608540d4731cee6399837174.

The ```DicomIOManager``` class always expected a directory as input, but was accepting both directories and files from ```cbica::ReadImage``` function, which was causing weird crashes in GDCM.

**Comment by ashishsingh18** [comment]
> 
> 
> DICOM Input handling has been updated courtesy [sarthakpati@207bbba](https://github.com/sarthakpati/CaPTk/commit/207bbbaedfd4bb5c608540d4731cee6399837174).
> 
> The `DicomIOManager` class always expected a directory as input, but was accepting both directories and files from `cbica::ReadImage` function, which was causing weird crashes in GDCM.

It was always the responsibility of the applications to pass the correct input(in this case a directory) to the DicomIOManager via cbica::ReadImage. We are already doing this from the GUI. Perhaps the CLI apps were never updated to pass the directories. However this should fix it.

**Comment by sarthakpati** [comment]
> > DICOM Input handling has been updated courtesy [sarthakpati@207bbba](https://github.com/sarthakpati/CaPTk/commit/207bbbaedfd4bb5c608540d4731cee6399837174).
> > The `DicomIOManager` class always expected a directory as input, but was accepting both directories and files from `cbica::ReadImage` function, which was causing weird crashes in GDCM.
> 
> It was always the responsibility of the applications to pass the correct input(in this case a directory) to the DicomIOManager via cbica::ReadImage. We are already doing this from the GUI. Perhaps the CLI apps were never updated to pass the directories. However this should fix it.

Perhaps I missed something, because I seem to remember having the discussion with you that the API for ```cbica::ReadImage``` should remain consistent and the DICOM handling should be done inside it, with the idea being any application would gain the DICOM functionality without having to change its behavior.. 

Also, passing a folder in an application could also potentially allude to performing batch processing for *all* images in that folder, which would cause confusion...

**Pull request merged**